plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/debug/ui/launchConfigurations/EnvironmentTab.java

```diff
@@ -0,0 +1,213 @@
+/******************************************************************************* 
+ * Copyright (c) 2021 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/
+package org.eclipse.reddeer.eclipse.debug.ui.launchConfigurations;
+
+import java.util.List;
+
+import org.eclipse.reddeer.common.exception.WaitTimeoutExpiredException;
+import org.eclipse.reddeer.common.wait.TimePeriod;
+import org.eclipse.reddeer.common.wait.WaitUntil;
+import org.eclipse.reddeer.swt.api.TableItem;
+import org.eclipse.reddeer.swt.condition.ShellIsAvailable;
+import org.eclipse.reddeer.swt.impl.button.OkButton;
+import org.eclipse.reddeer.swt.impl.button.PushButton;
+import org.eclipse.reddeer.swt.impl.button.RadioButton;
+import org.eclipse.reddeer.swt.impl.button.YesButton;
+import org.eclipse.reddeer.swt.impl.table.DefaultTable;
+import org.eclipse.reddeer.swt.impl.table.DefaultTableItem;
+import org.eclipse.reddeer.swt.impl.text.LabeledText;
+
+/**
+ * Represents Environment tab in Launch Configuration
+ * 
+ * @author Oleksii Korniienko olkornii@redhat.com
+ *
+ */
+public class EnvironmentTab extends LaunchConfigurationTab {
+
+	private static final String ADD_SHELL_TITLE = "New Environment Variable";
+	private static final String SELECT_SHELL_TITLE = "Select Environment Variables";
+	private static final String EDIT_SHELL_TITLE = "Edit Environment Variable";
+	private static final String OVERWRITE_SHELL_TITLE = "Overwrite variable?";
+
+	public EnvironmentTab() {
+		super("Environment");
+	}
+
+	/**
+	 * Return the list of all environment variables
+	 *
+	 * @return list of items
+	 */
+	public List<TableItem> getAllVariables() {
+		return new DefaultTable().getItems();
+	}
+
+	/**
+	 * Return the environment variable by name
+	 *
+	 * @return Environment Tab item
+	 */
+	public TableItem getVariable(String variableName) {
+		return new DefaultTable().getItem(variableName);
+	}
+
+	/**
+	 * Return the environment variable by id
+	 *
+	 * @return Environment Tab item
+	 */
+	public TableItem getVariable(int variableId) {
+		return new DefaultTable().getItem(variableId);
+	}
+
+	/**
+	 * Add new environment variable
+	 *
+	 * @param name  variable name
+	 * @param value variable value
+	 */
+	public void add(String name, String nalue) {
+		new PushButton("Add...").click();
+		new WaitUntil(new ShellIsAvailable(ADD_SHELL_TITLE));
+		new LabeledText("Name:").setText(name);
+		new LabeledText("Value:").setText(nalue);
+		new OkButton().click();
+		try {
+			new WaitUntil(new ShellIsAvailable(OVERWRITE_SHELL_TITLE), TimePeriod.SHORT);
+			new YesButton().click();
```

Comment:
Overwriting value should be disabled by default

Reply:
fixed

Reply:
I think that overwriting the value should be default choice but the option not to override is also implemented in api. I can imagine that user would like to add the intended value no matter there already exists one as he should know what he wants. Instead of not passing it be default which would require a debugging session to find out why the env is not set, imho.